plugins/restful/RestfulDataProviderEFQ.php

```diff
@@ -278,9 +278,13 @@ public function getTotalCount() {
   protected function addExtraInfoToQuery($query) {
     parent::addExtraInfoToQuery($query);
     $entity_type = $this->getEntityType();
-    // Add a generic entity access tag to the query.
-    $query->addTag($entity_type . '_access');
-    $query->addMetaData('restful_handler', $this);
+    // The only time you need to add the access tags to a EFQ is when you don't
+    // have fieldConditions.
+    if (empty($query->fieldConditions)) {
+      // Add a generic entity access tag to the query.
+      $query->addTag($entity_type . '_access');
```

Comment:
There is no documentation around this (security!) issue. But from what I gather you are not supposed to add access tags to entities when using fieldConditions. On the contrary you need to add the `DANGEROUS_ACCESS_CHECK_OPT_OUT` if you want to avoid them.

The `field_sql_storage` module will add the tag `entity_field_access` when doing a field condition.